### ISSUE TITLE
format doesn't react on extra characters

### ISSUE THREAD

**Issue by cheepo2109** [s-yadav/react-number-format#371]
Hey, thanks for the nice library.
I stumbled upon an issue with format prop.
Let's say I used code similar to the example

```
function limit(val, max) {
  if (val.length === 1 && val[0] > max[0]) {
    val = '0' + val;
  }

  if (val.length === 2) {
    if (Number(val) === 0) {
      val = '01';
  } else if (val > max) {
      val = max;
    }
  }

  return val;
}

export function cardExpiry(val) {
  const month = limit(val.substring(0, 2), '12');
  const year = val.substring(2, 4);

  return month + (year.length ? '/' + year : '');
}
<NumberFormat format={cardExpiry}/>

```
I noticed that this format function doesn't react on characters like comma,dot or slash, when I type them on the input. I want format function to adapt depending on these chars.
How can I make this function react to them? 

Thanks!

**Comment by cheepo2109** [comment]
Also I noticed when I enable custom format, all the other formats stop working :)

**Comment by s-yadav** [comment]
Yes, custom formatting leaves all the formatting responsibility to format function. 
Right now number-format trims all non-numeric character. Regarding supporting some custom characters, I will see if I can add a prop something like allowedChar

**Comment by RotaruAndrei1998** [comment]
> Yes, custom formatting leaves all the formatting responsibility to format function.
> Right now number-format trims all non-numeric character. Regarding supporting some custom characters, I will see if I can add a prop something like allowedChar

So, at this moment, do we have a solution for typing different characters and format the value after?
